packages/webapp/src/containers/Finances/saga.js

```diff
@@ -148,6 +151,69 @@ export function* getFarmExpenseTypeSaga() {
   }
 }
 
+export function* addCustomExpenseTypeSaga(action) {
+  const { expenseTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  const { expense_name } = action.custom_expense_type;
+
+  try {
+    const result = yield call(axios.post, expenseTypeUrl, { farm_id, expense_name }, header);
+    if (result) {
+      yield put(enqueueSuccessSnackbar(i18n.t('message:EXPENSE_TYPE.SUCCESS.ADD')));
+      yield call(getFarmExpenseTypeSaga);
+      history.push('/manage_custom_expenses');
+    }
+  } catch (e) {
+    console.log('failed to add new expense type to the database');
+    yield put(enqueueErrorSnackbar(i18n.t('message:EXPENSE_TYPE.ERROR.ADD')));
+  }
+}
+
+export function* updateCustomExpenseTypeSaga(action) {
+  const { expenseTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  const { expense_type_id } = action;
+  const { expense_type_name } = action.custom_expense_type;
+
+  try {
+    const result = yield call(
+      axios.put,
+      `${expenseTypeUrl}/${expense_type_id}`,
+      { farm_id, expense_type_name },
+      header,
+    );
+    if (result) {
+      yield put(enqueueSuccessSnackbar(i18n.t('message:EXPENSE_TYPE.SUCCESS.UPDATE')));
+      yield call(getFarmExpenseTypeSaga);
+      history.push('/manage_custom_expenses');
+    }
+  } catch (e) {
+    console.log('failed to update expense type in the database');
+    yield put(enqueueErrorSnackbar(i18n.t('message:EXPENSE_TYPE.ERROR.UPDATE')));
+  }
+}
+
+export function* retireCustomExpenseTypeSaga(action) {
+  const { expenseTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  const { expense_type_id } = action;
+
+  try {
+    const result = yield call(axios.delete, `${expenseTypeUrl}/${expense_type_id}`, header);
+    if (result) {
+      yield put(enqueueSuccessSnackbar(i18n.t('message:EXPENSE_TYPE.SUCCESS.DELETE')));
+      yield call(getFarmExpenseTypeSaga);
```

Comment:
Wow, it's so simple this way 😆 Just a fresh GET and full set every time!

Reply:
Just copying tasks haha. Might change though